plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -1154,21 +1196,27 @@ private ColumnMapping jsonColumnMapping()
         return ColumnMapping.sliceMapping(
                 jsonType,
                 (resultSet, columnIndex) -> jsonParse(utf8Slice(resultSet.getString(columnIndex))),
-                typedVarcharWriteFunction("json"),
+                typedVarcharWriteFunction("json", false),
                 DISABLE_PUSHDOWN);
     }
 
-    private static ColumnMapping typedVarcharColumnMapping(String jdbcTypeName)
+    private static ColumnMapping typedVarcharColumnMapping(String jdbcTypeName, boolean withCollate)
```

Comment:
drop `withCollate`, since it's (or: probably should be) always false

Reply:
It can be true if `jdbcTypeName` is `text`.